OCPBUGS-7229: OpenStack: Add load balancer Services limitations (#56004)
openshift/openshift-docs: modules/nw-osp-loadbalancer-etp-local.adoc

// Module included in the following assemblies:
// * networking/nw-osp-loadbalancer-limitations.adoc

:_content-type: CONCEPT
[id="nw-osp-loadbalancer-etp-local_{context}"]

> = `externalTrafficPolicy` set to `Local`

Reviewer: ```suggestion
= Local external traffic policies
```

> Setting `.spec.externalTrafficPolicy` on the load balancer Service makes sure to preserve source IP of the incoming traffic when it reaches Service endpoint pods. But if you're using Amphora Octavia provider, the source IP of the traffic will be replaced with the IP of the Amphora VM anyway. This problem isn't present when using OVN Octavia provider.

Reviewer: ```suggestion
You can set the  external traffic policy (ETP) parameter, `.spec.externalTrafficPolicy`, on a load balancer service to preserve the source IP address of incoming traffic when it reaches service endpoint pods. However, if your cluster uses the Amphora Octavia provider, the source IP of the traffic is replaced with the IP address of the Amphora VM. This behavior does not occur if your cluster uses the OVN Octavia provider.
```

> You can set the  external traffic policy (ETP) parameter, `.spec.externalTrafficPolicy`, on a load balancer service to preserve the source IP address of incoming traffic when it reaches service endpoint pods. However, if your cluster uses the Amphora Octavia provider, the source IP of the traffic is replaced with the IP address of the Amphora VM. This behavior does not occur if your cluster uses the OVN Octavia provider.

Reviewer: ```suggestion
You can set the external traffic policy (ETP) parameter, `.spec.externalTrafficPolicy`, on a load balancer service to preserve the source IP address of incoming traffic when it reaches service endpoint pods. However, if your cluster uses the Amphora Octavia provider, the source IP of the traffic is replaced with the IP address of the Amphora VM. This behavior does not occur if your cluster uses the OVN Octavia provider.
```

Author: Done!

> Having the `ETP` option set to `Local` requires that health monitors be created for the load balancer. Without health monitors, traffic can be routed to a node that doesn't have a functional endpoint, which causes the connection to drop. In order to force Cloud Provider OpenStack to create health monitors, you must set the value of the `create-monitor` option in the cloud provider configuration to `true`.

Reviewer 2: ```suggestion
Having the `ETP` option set to `Local` requires that health monitors be created for the load balancer. Without health monitors, traffic can be routed to a node that doesn't have a functional endpoint, which causes the connection to drop. To force Cloud Provider OpenStack to create health monitors, you must set the value of the `create-monitor` option in the cloud provider configuration to `true`.
```

In {rh-openstack} 16.1 and 16.2, the OVN Octavia provider does not support health monitors. Therefore, setting the ETP to local is unsupported.

> In {rh-openstack} 16.1 and 16.2, the Amphora Octavia provider does not support HTTP monitors on UDP pools. As a result, UDP load balancer services have `UDP-CONNECT` monitors created instead. Due to implementation details, this configuration only functions properly with the OVN-Kubernetes CNI plugin. When OpenShift SDN CNI plugin is used, the UDP services have alive nodes detected unreliably.

Reviewer 2: ```suggestion
In {rh-openstack} 16.1 and 16.2, the Amphora Octavia provider does not support HTTP monitors on UDP pools. As a result, UDP load balancer services have `UDP-CONNECT` monitors created instead. Due to implementation details, this configuration only functions properly with the OVN-Kubernetes CNI plugin. When the OpenShift SDN CNI plugin is used, the UDP services alive nodes are detected unreliably.
```

Reviewer 2: Not sure if my suggestion for the last part incorrectly changes the meaning; if so, please ignore.

Author: Nah, it makes perfect sense, thanks!
